nomad/fsm.go

```diff
@@ -555,6 +555,13 @@ func (n *nomadFSM) applyUpsertJob(buf []byte, index uint64) interface{} {
 		}
 	}
 
+	if req.Eval != nil {
```

Comment:
comment with something like:
```suggestion
	//COMPAT Prior to Nomad 0.12.x evaluations were submitted in a separate Raft log so this may be nil during server upgrades.
	if req.Eval != nil {
```